dxp/frontend_taglib_clay.md

```diff
@@ -0,0 +1,49 @@
+# What is `frontend-taglib-clay` and it's relation with clay-\*, @clayui, and css?
+
+## High-level:
+
+`frontend-taglib-clay` provides a collection of clay dependencies from npm and custom components to be used across DXP as taglibs and js dependencies. Versions of these npm modules are locked in this module, this is done to keep modules consistent across dxp. For example, by locking these versions, this means we use the same version of `@clayui/button` across all of dxp so that we can reduce the number of modules served with dxp.
```

Comment:
Let's always capitalize:

- "clay" → Clay
- "dxp" → DXP
- "npm" → NPM
- "js" → JS

I know you were worried about word count, so you could delete "For example," and just start that sentence with "By locking".

Reply:
@wincent  not something important, but I'm never 100% sure whether to use "NPM" or "npm", for example in their [about](https://www.npmjs.com/about) page they refer to it as npm in lowercase and if you do a quick Google search for npm you'll see this (which has npm, Npm and NPM) :shrug: 

![npm01](https://user-images.githubusercontent.com/5572/92363075-3aa68e80-f0f1-11ea-8527-0cd190cf5da8.png)

Reply:
🤦‍♂️ — yeah, seems they probably use "npm" more than they use the variants, and it looks like [their first commit in 2009](https://github.com/npm/cli/commit/4626dfa73b7847e9c42c1f799935f8242794d020) used "npm" already then.

I have historically used "NPM" for the company/registry/package format, and `npm` for the commandline tool, to distinguish them, but given what you posted I could be persuaded to do "npm" for both (sadly, means that there are quite a few of our repos with docs/references to "NPM" in them).

Reply:
> I have historically used "NPM" for the company/registry/package format, and npm for the commandline tool,

Makes sense to me.

Reply:
I think I use `npm` too. I usually check the top left header of each project I mention to see the capitalization, so:

- ![npm](https://www.npmjs.com/)
- ![webpack](https://webpack.js.org/e0b5805d423a4ec9473ee315250968b2.svg)

However, the rule doesn't work for Babel, which they call Babel, but capitalize as BABEL in the image :shrug: .

In any case, for npm the correct form seems to be npm. Even `npm, Inc.` is capitalized like that, and look at [their blog](https://blog.npmjs.org/)